File: examples/go/aws_example.go
package main

import (
	"os"

	"github.com/castai/pulumi-castai"
	"github.com/castai/pulumi-castai/aws"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

// runAwsExample shows how to connect an AWS EKS cluster to CAST AI
func runAwsExample() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		// Initialize the provider (API token will be read from environment variable CASTAI_API_TOKEN)
		provider, err := castai.NewProvider(ctx, "castai-provider", nil)
		if err != nil {
			return err
		}

		// Get AWS account ID from environment variable or use a default value
		accountID := os.Getenv("AWS_ACCOUNT_ID")
		if accountID == "" {
			accountID = "123456789012"
		}

		// Get AWS region from environment variable or use a default value
		region := os.Getenv("AWS_REGION")
		if region == "" {
			region = "us-west-2"
		}

		// Get EKS cluster name from environment variable or use a default value
		clusterName := os.Getenv("EKS_CLUSTER_NAME")
		if clusterName == "" {
			clusterName = "cast_ai_test_cluster"
		}

		// Create a connection to an EKS cluster
		eksArgs := &aws.EksClusterArgs{
			AccountId:              pulumi.String(accountID),
			Region:                 pulumi.String(region),
			EksClusterName:         pulumi.String(clusterName),
			DeleteNodesOnDisconnect: pulumi.Bool(true),
			// The following values need to be replaced with actual values from your AWS account
			// For demo purposes, we're using placeholder values
			SecurityGroupId:        pulumi.String("sg-12345678"),
			SubnetIds:              pulumi.StringArray{pulumi.String("subnet-12345678"), pulumi.String("subnet-87654321")},
		}

		eksCluster, err := aws.NewEksCluster(ctx, "eks-cluster-connection", eksArgs, pulumi.Provider(provider))
		if err != nil {
			return err
		}

		// Export the cluster ID
		ctx.Export("clusterId", eksCluster.ID())

		return nil
	})
}

File: examples/python/aws_example.py
"""
CAST AI AWS (EKS) Example for Python
This example demonstrates how to connect an AWS EKS cluster to CAST AI.
"""

import pulumi
import os
from pulumi_castai import Provider, EksCluster

# Check for required AWS credentials
required_vars = [
    "AWS_ACCESS_KEY_ID",
    "AWS_SECRET_ACCESS_KEY",
    "AWS_REGION",
    "CASTAI_API_TOKEN"
]

missing_vars = [var for var in required_vars if not os.environ.get(var)]
if missing_vars:
    print(f"Warning: Missing required AWS credentials: {', '.join(missing_vars)}")
    print("This is a simulation only - not creating actual resources.")

# Initialize the CAST AI provider
# You can set these via environment variables CASTAI_API_TOKEN and CASTAI_API_URL
api_token = os.environ.get("CASTAI_API_TOKEN", "your-api-token-here")
provider = Provider("castai-provider", api_token=api_token)

# Get AWS values from environment variables or use defaults
aws_region = os.environ.get("AWS_REGION", "us-west-2")
aws_account_id = os.environ.get("AWS_ACCOUNT_ID", "123456789012")

# Get EKS cluster name from environment variable or use a default value
eks_cluster_name = os.environ.get("EKS_CLUSTER_NAME", "cast_ai_test_cluster")

# Create a connection to an EKS cluster
eks_cluster = EksCluster("eks-cluster-connection",
    account_id=aws_account_id,           # AWS account ID
    region=aws_region,                   # AWS region
    eks_cluster_name=eks_cluster_name,   # EKS cluster name
    delete_nodes_on_disconnect=True,     # Remove nodes on disconnect
    
    # The following values need to be replaced with actual values from your AWS account
    # For demo purposes, we're using placeholder values
    security_group_id="sg-12345678",
    subnet_ids=["subnet-12345678", "subnet-87654321"],
    opts=pulumi.ResourceOptions(provider=provider)
)

# Export the cluster ID
pulumi.export("cluster_id", eks_cluster.id)
